mapper/dm_agent/src/error.rs

```diff
@@ -0,0 +1,32 @@
+use crate::collectd::CollectdError;
+use mqtt_client::Error as MqttClientError;
+use std::sync::Arc;
+use thin_edge_json::{
+    group::{MeasurementGrouper, MeasurementGrouperError},
+    serialize::ThinEdgeJsonSerializationError,
+};
+use tokio::sync::mpsc::error::SendError;
+
+#[derive(thiserror::Error, Debug)]
+pub enum DeviceMonitorError {
+    #[error(transparent)]
+    MqttClientError(#[from] Arc<MqttClientError>),
+
+    #[error(transparent)]
+    InvalidCollectdMeasurementError(#[from] CollectdError),
+
+    #[error(transparent)]
+    InvalidThinEdgeJsonError(#[from] MeasurementGrouperError),
+
+    #[error(transparent)]
+    ThinEdgeJsonSerializationError(#[from] ThinEdgeJsonSerializationError),
+
+    #[error(transparent)]
+    BatchingError(#[from] SendError<MeasurementGrouper>),
+}
+
+impl From<mqtt_client::Error> for DeviceMonitorError {
+    fn from(error: MqttClientError) -> Self {
+        Self::MqttClientError(Arc::new(error))
+    }
+}
```

Comment:
Since I was forced to have an `Arc<MqttClientError>` variant of the enum, I didn't want to create one more enum variant that wraps an owned `MqttClientError`. So used this trick to keep just the `Arc` variant.